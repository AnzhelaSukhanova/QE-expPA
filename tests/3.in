(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (and (and (or (bvult y (bvmul x #b0110)) (= y (bvmul x #b0110))) (or (bvult y (bvmul x #b0011)) (= y (bvmul x #b0011)))) (or (bvult y (bvmul x #b0010)) (= y (bvmul x #b0010))))))
