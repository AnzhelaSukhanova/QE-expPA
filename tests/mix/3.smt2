(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(declare-fun z () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4)))
 (or
 (or (bvult (bvshl #b0001 x) (bvadd y (bvadd (bvmul z #b0011) #b1000))) (= (bvshl #b0001 x) (bvadd y (bvadd (bvmul z #b0011) #b1000)))) 
 (or 
 (or (bvult (bvshl #b0001 x) (bvadd (bvshl #b0001 y) (bvadd (bvmul y #b1011) #b0100))) (= (bvshl #b0001 x) (bvadd (bvshl #b0001 y) (bvadd (bvmul y #b1011) #b0100))))
 (or 
 (or (bvult (bvmul (bvshl #b0001 y) #b0011) x) (= (bvmul (bvshl #b0001 y) #b0011) x)) 
 (or (bvult x (bvmul y #b1100)) (= x (bvmul y #b1100)))))))) 
