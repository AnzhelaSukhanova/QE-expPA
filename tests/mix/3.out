(let (($e1 (bvadd y (bvadd (bvmul z #b0011) #b1000)))) (let (($e2 (bvadd (bvshl #b0001 y) (bvadd #b0100 (bvmul y #b1011))))) (let (($e3 (= $e1 #b0000))) (let (($e4 (ite (bvult $e1 #b1000) (ite (bvult $e1 #b0100) (ite (bvult $e1 #b0010) (ite $e3 #b1111 #b0000) #b0001) #b0010) #b0011))) (let (($e5 (bvshl #b0001 $e4))) (let (($e6 (bvshl #b0001 (bvadd #b0001 $e4)))) (let (($e7 (= #b0001 $e1))) (let (($e8 (= $e1 #b0010))) (let (($e9 (= $e1 #b0100))) (let (($e10 (= #b1000 $e1))) (let (($e11 (bvult #b0001 $e1))) (let (($e12 (bvult #b0010 $e1))) (let (($e13 (bvult #b0100 $e1))) (let (($e14 (bvult #b1000 $e1))) (let (($e15 (= $e2 #b0000))) (let (($e16 (ite (bvult $e2 #b1000) (ite (bvult $e2 #b0100) (ite (bvult $e2 #b0010) (ite $e15 #b1111 #b0000) #b0001) #b0010) #b0011))) (let (($e17 (bvshl #b0001 $e16))) (let (($e18 (bvshl #b0001 (bvadd #b0001 $e16)))) (let (($e19 (= #b0001 $e2))) (let (($e20 (= $e2 #b0010))) (let (($e21 (= #b0100 $e2))) (let (($e22 (= #b1000 $e2))) (let (($e23 (bvult #b0001 $e2))) (let (($e24 (bvult #b0010 $e2))) (let (($e25 (bvult #b0100 $e2))) (let (($e26 (bvult #b1000 $e2))) (not (and (bvult (bvmul y #b1100) (bvshl #b0011 y)) (not (and (not (and (not $e24) (not $e23) (not $e20) (not $e19) (not $e25) (not $e21) (not $e26) (not $e22) $e15 (not $e15) (not $e23) (not $e19) (not $e24) (not $e20) (not $e25) (not $e21) (not $e26) (not $e22) $e15 (not $e15) (not $e23) (not $e19) (not (bvult $e18 $e2)) (not (bvult $e17 $e2)) (not $e24) (not (= $e2 $e18)) (not (= $e2 $e17)) (not $e20))) (not (and (not $e12) (not $e11) (not $e8) (not $e7) (not $e13) (not $e9) (not $e14) (not $e10) $e3 (not $e3) (not $e11) (not $e7) (not $e12) (not $e8) (not $e13) (not $e9) (not $e14) (not $e10) $e3 (not $e3) (not $e11) (not $e7) (not (bvult $e6 $e1)) (not (bvult $e5 $e1)) (not $e12) (not (= $e1 $e6)) (not (= $e1 $e5)) (not $e8)))))))))))))))))))))))))))))))))
